org.eclipse.scanning.event/src/org/eclipse/scanning/event/ConsumerImpl.java

```diff
@@ -423,79 +458,128 @@ public void run() throws EventException {
+		// NOTE: Not all consumers check the submit queue and
+		// pause before they start.
+        checkStartPaused();
+		
+		// It is possible to call start() and then awaitStart().
+		if (latchStart!=null) latchStart.countDown();
 		while(isActive()) {
         	try {
-        		checkPaused(); // blocks until not paused.
-        		if (!isActive()) break; // Might have pasued for a long time.
-        		
-        		// Consumes messages from the queue.
-	        	Message m = getMessage(uri, getSubmitQueueName());
-	            if (m!=null) {
-		        	waitTime = 0; // We got a message
-	            	
-	            	// TODO FIXME Check if we have the max number of processes
-	            	// exceeded and wait until we don't...
-	            	
-	            	TextMessage t = (TextMessage)m;
-	            	
-	            	final String json  = t.getText();
-	            	
-					@SuppressWarnings("unchecked")
-					final U bean   = (U) service.unmarshal(json, getBeanClass());
-                    
-	            	executeBean(bean);
-	            	
-	            }
+	            boolean ok = consume();
+	            if (!ok) break;
 	            
         	} catch (EventException | InterruptedException ne) {
         		if (Thread.interrupted()) break;
  				logger.error("Cannot consume message ", ne);
-       		    if (isDurable()) continue;
+       		    if (isDurable()) {
+       		    	continue;
+       		    }
         		break;
          		
         	} catch (Throwable ne) {
-        		logger.debug("Error in consumer!", ne);
-        		if (ne.getClass().getSimpleName().contains("Json")) {
-            		logger.error("Fatal except deserializing object!", ne);
-            		continue;
-        		}
-        		if (ne.getClass().getSimpleName().endsWith("UnrecognizedPropertyException")) {
-        			logger.error("Cannot deserialize bean!", ne);
-            		continue;
-        		}
-        		
-        		if (ne.getClass().getSimpleName().endsWith("ClassCastException")) {
-    				logger.error("Problem with serialization?", ne);
+        		boolean ok = manageError(ne);
+        		if (ok) {
+        			continue;
+        		} else {
+        			break;
         		}
-
-        		
-        		if (!isDurable()) break;
-        		        		
-       			try {
-       				if (Thread.interrupted()) break;
-					Thread.sleep(Constants.getNotificationFrequency());
-				} catch (InterruptedException e) {
-					throw new EventException("The consumer was unable to wait!", e);
-				}
-       			
-       			waitTime+=Constants.getNotificationFrequency();
-    			checkTime(waitTime); 
-    			
-        		logger.warn(getName()+" ActiveMQ connection to "+uri+" lost.");
-        		logger.warn("We will check every 2 seconds for 24 hours, until it comes back.");
-
-        		continue;
         	}
 
 		}
 	}
-	
+
+	private boolean consume() throws Exception {
+		
+		checkPaused(); // blocks until not paused.
+		if (!isActive()) return false; // Might have pasued for a long time.
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') At most one statement is allowed per line, but 2 statements were found on this line. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS00122)